consensus/bor/api.go

```diff
@@ -75,13 +75,37 @@ func rankMapDifficulties(values map[common.Address]uint64) []difficultiesKV {
 }
 
 // GetSnapshotProposerSequence retrieves the in-turn signers of all sprints in a span
-func (api *API) GetSnapshotProposerSequence(number *rpc.BlockNumber) (BlockSigners, error) {
-	snapNumber := *number - 1
+func (api *API) GetSnapshotProposerSequence(blockNrOrHash *rpc.BlockNumberOrHash) (BlockSigners, error) {
+	var (
+		snap       *Snapshot
+		snapNumber rpc.BlockNumber
+		header     *types.Header
+		err        error
+	)
+
+	//nolint:nestif
+	if blockNrOrHash == nil {
+		header = api.chain.CurrentHeader()
+		snapNumber = rpc.BlockNumber(header.Number.Int64() - 1)
+		snap, err = api.GetSnapshot(&snapNumber)
+	} else {
+		if blockNr, ok := blockNrOrHash.Number(); ok {
+			snapNumber = blockNr - 1
+			if blockNr == rpc.LatestBlockNumber {
```

Comment:
I believe this is done because rpc.LatestBlock is stored as (-1). Post milestone we might provide the "finalised" tag as well. Should we update this check to something like `if blockNr <= 0`?

Reply:
We could sure , though if someone passes `PendingBlockNumber` (-2) for instance we might end up giving an inaccurate response.

Reply:
Alright, for now just checking for Latest sounds good.